### ISSUE TITLE
 All apps are here but can't open?

### ISSUE THREAD

**Issue by Edwin1212256756** [winapps-org/winapps#104]
![image](https://github.com/winapps-org/winapps/assets/85831668/4536ecd8-e402-414e-8b4e-7cae553269ed)

I can see all my apps but when i launch any of them i get this error:

![image](https://github.com/winapps-org/winapps/assets/85831668/9796c981-67c5-4754-869b-6a4b25346030)

I got explorer to run with: bin/winapps check but i can't open explorer in the men.

**Comment by oskardotglobal** [comment]
Please symlink the winapps executable to ~/.local/bin/winapps

**Comment by oskardotglobal** [comment]
Also duplicate of #89, but gonna keep this issue as you actually show an error message here

**Comment by oskardotglobal** [comment]
Furthermore, please make sure you're actually following our installation guide and have our version of the source instead of the abandoned version by fmstrat

**Comment by Edwin1212256756** [comment]
Now that i've reinstalled my whole linux, i followed the new guide made by y'all.

got the VM working with docker but when i run the check command i get this output:
    ~/winapps    main !1  bin/winapps check                                                        ✔ 
[12:14:30:292] [14538:14539] [WARN][com.freerdp.crypto] - Certificate verification failure 'self-signed certificate (18)' at stack position 0
[12:14:30:292] [14538:14539] [WARN][com.freerdp.crypto] - CN = DOCKERW-NUS865N
[12:14:32:506] [14538:14539] [INFO][com.freerdp.gdi] - Local framebuffer format  PIXEL_FORMAT_BGRX32
[12:14:32:506] [14538:14539] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_BGRA32
[12:14:33:534] [14538:14553] [INFO][com.freerdp.channels.rdpdr.client] - Loading device service drive [home] (static)
[12:14:33:534] [14538:14539] [INFO][com.freerdp.channels.rdpsnd.client] - [static] Loaded fake backend for rdpsnd
[12:14:33:535] [14538:14539] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel rdpgfx
[12:14:33:535] [14538:14539] [INFO][com.freerdp.channels.drdynvc.client] - Loading Dynamic Virtual Channel disp
[12:14:37:920] [14538:14539] [WARN][com.freerdp.client.x11] - xf_Pointer: Invalid appWindow
[12:14:37:923] [14538:14539] [WARN][com.freerdp.client.x11] - xf_Pointer: Invalid appWindow
[12:14:37:924] [14538:14539] [WARN][com.freerdp.client.x11] - xf_Pointer_Set: handle=0
[12:14:38:483] [14538:14539] [INFO][com.freerdp.client.x11] - Logon Error Info LOGON_FAILED_OTHER [LOGON_MSG_BUMP_OPTIONS]
[12:14:38:493] [14538:14557] [ERROR][com.freerdp.utils] - Caught signal 'Segmentation fault' [11]
[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 0: /usr/lib/libwinpr2.so.2(winpr_backtrace+0x55) [0x72fa41bdd4b5]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 1: /usr/lib/libwinpr2.so.2(winpr_log_backtrace_ex+0x35) [0x72fa41bdf3d5]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 2: /usr/lib/libfreerdp2.so.2(+0x14244) [0x72fa41d8e244]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 3: /usr/lib/libc.so.6(+0x3ce20) [0x72fa41a00e20]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 4: xfreerdp(+0x172c5) [0x5b42ef85d2c5]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 5: xfreerdp(+0x17cf3) [0x5b42ef85dcf3]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 6: /usr/lib/libfreerdp2.so.2(+0x2b9fc) [0x72fa41da59fc]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 7: /usr/lib/libfreerdp2.so.2(+0x21bcf) [0x72fa41d9bbcf]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 8: /usr/lib/libfreerdp-client2.so.2(+0x4049c) [0x72fa41f5749c]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 9: /usr/lib/libfreerdp-client2.so.2(+0x42933) [0x72fa41f59933]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 10: /usr/lib/libfreerdp-client2.so.2(+0x1a820) [0x72fa41f31820]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 11: /usr/lib/libwinpr2.so.2(+0x4ec80) [0x72fa41bf8c80]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 12: /usr/lib/libc.so.6(+0x8e1cf) [0x72fa41a521cf]

[12:14:38:494] [14538:14557] [ERROR][com.freerdp.utils] - 13: /usr/lib/libc.so.6(+0x10f6ec) [0x72fa41ad36ec]

bin/winapps: line 128: 14538 Segmentation fault      (core dumped) $FREERDP_COMMAND /d:"$RDP_DOMAIN" /u:"$RDP_USER" /p:"$RDP_PASS" /scale:$RDP_SCALE +auto-reconnect +home-drive -wallpaper +dynamic-resolution $MULTI_FLAG /app:program:"explorer.exe" /v:"$RDP_IP"

**Comment by oskardotglobal** [comment]
Please install freerdp3 and uninstall freerdp2.

**Comment by Edwin1212256756** [comment]
you need bc! ?

**Comment by Edwin1212256756** [comment]
   ~/winapps    main !1  ./installer.sh                                                                              ✔ 
? Would you like to install for the current user or the whole system? User
Removing any old configurations...
./installer.sh: line 297: : command not found
./installer.sh: line 298: : command not found
Installing...
  Checking for installed apps in RDP machine (this may take a while)..../installer.sh: line 50: /home/edwin/winapps/apps//home/edwin/winapps/apps/info: No such file or directory

**Comment by Edwin1212256756** [comment]
./installer.sh                                                  
? Would you like to install for the current user or the whole system? System
Removing any old configurations...
Installing...
  Checking for installed apps in RDP machine (this may take a while)..../installer.sh: line 48: /home/edwin/winapps/apps/info: No such file or directory
 Finished.

The RDP connection failed to connect or run. Please confirm FreeRDP can connect with:
  bin/winapps check

If it cannot connect, this is most likely due to:
  - You need to accept the security cert the first time you connect (with 'check')
  - Not enabling RDP in the Windows VM
  - Not being able to connect to the IP of the VM
  - Incorrect user credentials in winapps.conf
  - Not merging install/RDPApps.reg into the VM

But the test command works fine?

**Comment by oskardotglobal** [comment]
@Edwin1212256756 Please try the latest commits in that PR

**Comment by Edwin1212256756** [comment]
Still same issue, but before the error about the vm not running, it opens the vm like the login screen and it logs and and closes and opens a command promt for like 0.1 second and closes and after a while the vm not running error comes in the console.

**Comment by oskardotglobal** [comment]
Are you running freerdp3?

**Comment by Edwin1212256756** [comment]
think so, freerdp outputs nothing because it's not installed, but xfreerdp3 works so i guess so

**Comment by Edwin1212256756** [comment]
once i start the install script, i can see xfreerdp3 starts to life


![image](https://github.com/winapps-org/winapps/assets/85831668/a244fe1d-a2e0-4288-aca2-617222fe3b95)

**Comment by Alchemi1963** [comment]
I am experiencing these same issues.
Confirmed the freerdp version is 3.5.1

I'd also like the add that bin/winapps manual "C:\Program Files\Microsoft Office\root\Office16\WINWORD.exe" _does_ work.

**Comment by oskardotglobal** [comment]
Yeah, freerdp on wayland seems like the culprit here, did you try changing xfreerdp3 to wlfreerdp (or whatever it's called on your distro) in the config?

**Comment by Alchemi1963** [comment]
Update:
it is **not** wayland related.
I tried wlfreerdp3 but it failed completely, xfreerdp3 works as intented (I tried inputting the command installer.sh uses and it worked, except for the cmd part)

I found that since installer.sh uses printf, the lines aren't written to the installed.bat file as intended which breaks it.
This is the case on at least on my installation (ArchLinux - GNU coreutils v9.5).

I suspect this is also the case for @Edwin1212256756 .
This also explains why @Edwin1212256756 saw freerdp3 running, but nothing else. And why I did see cmd open for a brief second and exit (it's because it didn't do anything because of the broken command.

The broken changes of printf are that \\\\ is printed as \ and newlines aren't a thing.
This means every command is directly followed by the next one and the UNC path begins with \tsclient instead of \\tsclient
